### PR TITLE
fix: ingester min-unpersisted-sequence-number calc + doc

### PR DESCRIPTION
**Pull request by crepererum**

### CODE REVIEW

**Comment by NGA-TRAN** [comment]
@pauldix or @domodwyer should review this. I do not follow much how this works for replay

**Comment by alamb** [comment]
I agree I don't feel qualified to review this and would feel better with @pauldix  or @domodwyer  giving it a review

File: ingester/src/lifecycle.rs [hunk 1]
```diff
@@ -359,12 +359,21 @@ impl LifecycleManager {
...
+            //   persisted. Note that can cannot use "max + 1" because the lifecycle handler receives writes on a
+            //   partition level and therefore might run while data for a single sequence number is added. So the max
+            //   sequence number that we have just persisted might have more data.
             for (sequencer_id, sequence_number) in sequencer_maxes {
                 let min = rest
                     .iter()
                     .filter(|s| s.sequencer_id == sequencer_id)
-                    .max_by_key(|s| s.first_sequence_number)
                     .map(|s| s.first_sequence_number)
+                    .min()
```
Comment (alamb) [inline ingester/src/lifecycle.rs:376]: I think this code does what the comments say it does, but I I am not 100% what it says is what it should do

**Review by pauldix — APPROVED** [review]
LGTM

**Pull request merged**